### ISSUE TITLE
v2.3.0 Update Drops/Recreates API

### ISSUE THREAD

**Issue by SethO** [sid88in/serverless-appsync-plugin#607]
I just reproduced something I saw on one of my test APIs - updating this library to v2.3.0 will cause your API to be deleted/recreated. This gives it a new ApiId, URL, the works. I don't think this was intended. For any folks who are sharing the ApiId as a string/constant, this will wreck them.

![image](https://github.com/sid88in/serverless-appsync-plugin/assets/1322700/96149eda-6ea0-4b87-aae1-8e61bf879baa)
![image](https://github.com/sid88in/serverless-appsync-plugin/assets/1322700/d869eba1-1905-46ad-8619-9e126d0b7074)

As a side note, my project with a Merged API was saved from this because, evidently, you cannot delete a source API if it is used by a Merged. That saved my bacon.
![image](https://github.com/sid88in/serverless-appsync-plugin/assets/1322700/d5fdd5b5-d57a-4b00-a427-4e258d5b379b)

**Comment by bboure** [comment]
Oh no! This isn't good.

I'm pretty sure it has been introduced by #604 
I experienced the same behavior on one of my test APIs when I tested it, but I thought it was because I was switching from `GLOBAL` to `PRIVATE` and vice-versa.

I will try to confirm this issue and come up with a fix asap.

The problem might be that we should have kept `undefined` [here](https://github.com/sid88in/serverless-appsync-plugin/blob/master/src/resources/Api.ts#L77) 
 when `visibility` is not specified
